tests/cpp/test_gpu3.cpp

```diff
@@ -8051,23 +8051,30 @@ TEST_F(NVFuserTest, AvoidCachingSliceInput) {
 
   FusionExecutorCache executor_cache(std::move(fusion));
   auto cg_outputs = executor_cache.runFusionWithInputs(inputs);
-  // check segment and sliced tvs are not cached
+  // check segmentation and sliced tvs are not cached if not scheduled by
+  // the resize scheduler
   auto kernel_runtime = executor_cache.getMostRecentKernelRuntime();
-  NVF_CHECK(kernel_runtime->isSegmented(), "segmentation didn't happen");
+  ASSERT_TRUE(kernel_runtime->isSegmented()) << "segmentation didn't happen";
   const auto num_segments = kernel_runtime->fusionSegments()->groups().size();
-  NVF_CHECK(num_segments == 3, "Expect 3 segments, got: ", num_segments);
-  for (const auto& exec : kernel_runtime->executors()) {
+  EXPECT_EQ(num_segments, 2) << "Expect 2 segments, got: " << num_segments;
+  for (const auto i : c10::irange(kernel_runtime->executors().size())) {
+    const auto& exec = kernel_runtime->executors().at(i);
     if (!exec->isA<KernelExecutor>()) {
       continue;
     }
+    if (kernel_runtime->schedulerHeuristics()
+            ->heuristicsList()
+            .at(i)
+            ->scheduler_type == SchedulerType::Resize) {
+      continue;
+    }
     const auto* ke = exec->as<KernelExecutor>();
     for (auto expr : ke->fusion()->exprs()) {
       if (expr->isA<SliceOp>()) {
         auto slice = expr->as<SliceOp>();
-        NVF_CHECK(
-            slice->in()->getMemoryType() == MemoryType::Global,
-            "slice input must be in global memory, get: ",
-            slice->in()->getMemoryType());
+        EXPECT_EQ(slice->in()->getMemoryType(), MemoryType::Global)
```

Comment:
I'm not sure if this check still makes sense with this PR. Don't you want all slices picked up by the new resize scheduler?

Reply:
I thought not all segments may be picked up by the resize scheduler yet, but let me check.

Reply:
This fusion is now segmented as:

```
group details:
g{(resize)
group id: 0
inputs:
  T1_g___half[iS2{i3}] __half
outputs:
  T11_g_float[bS26{1}, iS61{4096}] float
  T17_g___half[iS44{8}rf, iS45{( ceilDiv(i3, 8) )}rf] __half


T17_g___half[iS44{8}rf, iS45{( ceilDiv(i3, 8) )}rf] = view( T1_g___half[iS2{i3}] )
(21)
T8_g___half[iS56{8}, iS18{512}rf]
   = slice( T17_g___half[iS44{8}rf, iS45{( ceilDiv(i3, 8) )}rf], { {0, 8, 1} {0, 512, 1} } )
(22)
T18_g___half[iS59{4096}rf] = view( T8_g___half[iS56{8}, iS18{512}rf] )
(25)
T10_g___half[bS24{1}, iS60{4096}]
   = broadcast( T18_g___half[iS59{4096}rf] )
(26)
T11_g_float[bS26{1}, iS61{4096}]
   = __half2float(T10_g___half[bS24{1}, iS60{4096}]);
(11)
i97 = ceilDiv(i3, 8);
(19)
}

g{(inner_persistent)
group id: 1
inputs:
  T0_g___half[iS0{i0}, iS68{4096}] __half
  T17_g___half[iS44{8}rf, iS45{( ceilDiv(i3, 8) )}rf] __half
outputs:
  T16_g_float[iS40{i0}, iS41{4096}] float


T2_l_float[iS3{i0}, iS69{4096}]
   = __half2float(T0_g___half[iS0{i0}, iS68{4096}]);
(0)
T3_g_float[iS5{i0}, iS70{4096}]
   = expf(T2_l_float[iS3{i0}, iS69{4096}]);
(1)
T4_l_float[iS7{i0}, rS71{4096}]
   = reduction( T3_g_float[iS5{i0}, iS70{4096}], op = add, initial value = float(0), allreduce = false )
(2)
T5_g_float[iS9{i0}, bS10{1}]
   = broadcast( T4_l_float[iS7{i0}, rS71{4096}] )
(3)
T6_g_float[iS11{i0}, iS72{4096}]
   = T3_g_float[iS5{i0}, iS70{4096}]
   / T5_g_float[iS9{i0}, bS10{1}];
(4)
T12_g___half[iS62{8}, iS30{512}rf]
   = slice( T17_g___half[iS44{8}rf, iS45{( ceilDiv(i3, 8) )}rf], { {0, 8, 1} {1536, 2048, 1} } )
(23)
T19_l___half[iS65{4096}rf] = view( T12_g___half[iS62{8}, iS30{512}rf] )
(28)
T14_g___half[bS36{1}, iS66{4096}]
   = broadcast( T19_l___half[iS65{4096}rf] )
(29)
T15_g_float[bS38{1}, iS67{4096}]
   = __half2float(T14_g___half[bS36{1}, iS66{4096}]);
(17)
T16_g_float[iS40{i0}, iS41{4096}]
   = T6_g_float[iS11{i0}, iS72{4096}]
   * T15_g_float[bS38{1}, iS67{4096}];
(18)
}
```

The inner-persistent segment also has a slice, so the check still makes for that segment.

And looks like there's some inefficient segmentation here. `T17` is the output of the first reshape, which is consumed by the two slices. The reshape is done by the first segment, and the reshape output is an output of the first segment. Instead, the second segment should just load `T1` again and do the same reshape.

It's probably related to the uop recomputation. It seems we should consider reshape as a uop for the purpose of segmentation optimization. CC: @liqiangxl